src/traces/heatmap/make_bound_array.js

```diff
@@ -0,0 +1,78 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Registry = require('../../registry');
+
+module.exports = function makeBoundArray(trace, arrayIn, v0In, dvIn, numbricks, ax) {
+    var arrayOut = [],
+        isContour = Registry.traceIs(trace, 'contour'),
```

Comment:
non- :no_entry: , but it would be nice to remove those `Registry.traceIs` calls in `makeBoundArray` and have two (many more) boolean options instead: `withCenteredBricks` (true for contour and gl2d traces) and `startAtZero` (for histogram2d traces and categorical axes)

That way `makeBoundArray` would no longer need `trace` as an argument